tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs

```diff
@@ -447,82 +532,189 @@ public static void TestSpanDangerousGetPinnableReferenceByte(int length)
-            byte temp;
-            foreach (var iteration in Benchmark.Iterations)
-                using (iteration.StartMeasurement())
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                            temp = span[length/2];
+            InvokeTestSpanIndexHoistable<byte>(length);
+        }
+
+        [Benchmark(InnerIterationCount = BaseIterations)]
+        [InlineData(1)]
+        [InlineData(10)]
+        [InlineData(100)]
+        [InlineData(1000)]
+        public static void TestSpanIndexHoistableString(int length)
+        {
+            InvokeTestSpanIndexHoistable<string>(length);
+        }
+
+        static void InvokeTestSpanIndexHoistable<T>(int length)
+        {
+            var array = new T[length];
+
+            Invoke((int innerIterationCount) => TestSpanIndexHoistable<T>(array, length, innerIterationCount),
+                "TestSpanIndexHoistable<{0}>({1})", typeof(T).Name, length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void TestSpanIndexHoistable<T>(T[] array, int length, int iterationCount)
+        {
+            var sink = Sink<T>.Instance;
+            var span = new Span<T>(array);
+
+            for (int i = 0; i < iterationCount; i++)
+                    sink.Data = span[length/2];
+        }
+        #endregion
 
+        #region TestArrayIndexHoistable<T>
+        [Benchmark(InnerIterationCount = BaseIterations)]
+        [InlineData(1)]
+        [InlineData(10)]
+        [InlineData(100)]
+        [InlineData(1000)]
+        public static void TestArrayIndexHoistableByte(int length)
+        {
+            InvokeTestArrayIndexHoistable<byte>(length);
         }
 
         [Benchmark(InnerIterationCount = BaseIterations)]
         [InlineData(1)]
         [InlineData(10)]
         [InlineData(100)]
         [InlineData(1000)]
-        public static void TestSpanIndexString(int length)
+        public static void TestArrayIndexHoistableString(int length)
         {
-            var array = new string[length];
-            var span = new Span<string>(array);
-            string temp;
-            foreach (var iteration in Benchmark.Iterations)
-                using (iteration.StartMeasurement())
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                        temp = span[length / 2];
+            InvokeTestArrayIndexHoistable<string>(length);
+        }
+
+        static void InvokeTestArrayIndexHoistable<T>(int length)
+        {
+            var array = new T[length];
+            Invoke((int innerIterationCount) => TestArrayIndexHoistable<T>(array, length, innerIterationCount),
+                "TestArrayIndexHoistable<{0}>({1})", typeof(T).Name, length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void TestArrayIndexHoistable<T>(T[] array, int length, int iterationCount)
+        {
+            var sink = Sink<T>.Instance;
```

Comment:
I believe so.  Any type initialization should get triggered during the warm-up runs, so then the measured code will just verify that the type is initialized and load the field.